### PR TITLE
test(desktop): lock onboarding starter-prompt capability contract

### PR DESCRIPTION
**Pull request by eulicesl**
## Summary

- **Problem:** The post-onboarding "Try asking…" starter suggestions are answered by the **default piMono chat**, which is backed by Omi's 7 tools (memories, action items, goals, screen history) and *cannot* read files, run shell commands, or inspect code. A suggestion that promised any of those would return an empty result and read to the user as the app being broken.
- **What changed:** No copy change was needed — the existing suggestions are all already answerable. This **locks that as a contract**: the suggestion strings are extracted to `universalSuggestion` + `allKnownSuggestions` constants (so the test target sees exactly what ships), and four tests guard against future drift.
- **What did NOT change (scope boundary):** No behavioral change to which suggestions appear. No empty-state hero / chip redesign. No mode-aware suggestion sets (the chat empty-state copy was audited and is already accurate).

## Linked Issue / Bounty

- N/A — independent contribution.

## Root Cause (bug fixes only)

- N/A — this is a guardrail/test PR, not a bug fix. It prevents a future regression rather than fixing a current one.

## How It Works

- `OnboardingPromptSuggestionBuilder.build(from:)` now references two static constants instead of inline string literals: `universalSuggestion` (the always-first opener) and `allKnownSuggestions` (the full vocabulary in declaration order). This makes the exact strings the product emits visible to the test target.
- Four new `ChatDiscoverabilityTests` lock the contract:
  - no suggestion contains a blocked capability keyword (`code`, `file`, `bash`, `shell`, `terminal`, `github`);
  - the universal opener is always first;
  - the vocabulary stays within `build()`'s `.prefix(6)` cap (entries beyond 6 would be silently unreachable);
  - no duplicates (which `build()`'s `NSOrderedSet` dedup would silently collapse).

## Change Type

- [x] Chore / infra (test guardrail)

## Scope

- [x] Desktop app (Swift/macOS)

## Security Impact

- New permissions or capabilities introduced? **No**
- Auth or token handling changed? **No**
- Data access scope changed (screen data, OCR, user data)? **No**
- New or changed network calls? **No**
- Plugin or tool execution surface changed? **No**

## Testing

- [x] Built locally — `xcrun swift build -c debug --package-path Desktop` → `Build complete!`
- [x] Manual verification: compile + test run on macOS.
- [x] Existing tests pass
- [x] New tests added — 4 new cases in `ChatDiscoverabilityTests` (17 total in the suite, all green).

## Human Verification

- Verified scenarios: clean debug build against current `main`; full `ChatDiscoverabilityTests` suite passes (13 pre-existing + 4 new).
- Edge cases checked (via tests): a suggestion reaching for file/code/shell capability fails CI; vocabulary cap and dedup invariants hold.
- What I did **not** verify (and why): this is a pure refactor-to-constants plus tests, with no user-visible behavior change, so no runtime UI verification was needed.

## Evidence

- [x] Test output:
  ```
  Test Suite 'ChatDiscoverabilityTests' passed — Executed 17 tests, 0 failures
  Build complete!
  ```

## Docs

- [x] N/A — no docs impact

## Performance, Privacy, and Reliability

- No runtime impact (string constants + tests). Reliability: prevents a class of "the AI is dumb" empty-result complaints by keeping starter suggestions within what the default chat can actually answer.

## Migration / Backward Compatibility

- Backward compatible? **Yes**
- Migration needed? **No** — same suggestions emitted; only their source representation changed.

## Risks and Mitigations

- **Risk:** The blocked-keyword scan could false-positive on a legitimate future suggestion (e.g. one that mentions "file" innocuously). **Mitigation:** the keyword list is intentionally narrow, and a false positive surfaces at CI time as an obvious, easily-adjusted test failure rather than shipping.

## AI Disclosure

- Tools used: Claude Code (Claude Opus).
- AI-assisted portions: the refactor, the tests, and this PR description.
- How I verified correctness: ran the local debug build and the full `ChatDiscoverabilityTests` suite (all green); confirmed the emitted suggestions are unchanged from before.

### CODE REVIEW

File: desktop/Desktop/Sources/OnboardingPromptSuggestions.swift [hunk 1]
```diff
@@ -35,28 +35,64 @@ enum PostOnboardingPromptSuggestions {
...
 
     if !coordinator.goalDraft.isEmpty {
-      suggestions.append("Break my goal into the next 3 steps.")
+      suggestions.append(allKnownSuggestions[3])
     }
 
-    suggestions.append("What on my screen matters most right now?")
-    suggestions.append("What's the highest-leverage thing I can do next?")
+    suggestions.append(allKnownSuggestions[4])
+    suggestions.append(allKnownSuggestions[5])
```
[the remark this answers is not in the document]
Comment (eulicesl) [reply desktop/Desktop/Sources/OnboardingPromptSuggestions.swift]: Fixed in fac5244a1. `build()` no longer indexes into the array at all — each suggestion is a named constant (`emailSuggestion`, `screenSuggestion`, …) and `build()` references those directly, so there's no `[4]`/`[5]` access to go out of bounds if the vocabulary shrinks. Also tightened the bound test to `== 6`.

File: desktop/Desktop/Sources/OnboardingPromptSuggestions.swift [hunk 2]
```diff
@@ -35,28 +35,64 @@ enum PostOnboardingPromptSuggestions {
...
 
     if !coordinator.calendarSummary.isEmpty {
-      suggestions.append("Where can I find focus time this week?")
+      suggestions.append(allKnownSuggestions[2])
     }
 
     if !coordinator.goalDraft.isEmpty {
-      suggestions.append("Break my goal into the next 3 steps.")
+      suggestions.append(allKnownSuggestions[3])
     }
```
[the remark this answers is not in the document]
Comment (eulicesl) [reply desktop/Desktop/Sources/OnboardingPromptSuggestions.swift:110]: Fixed in fac5244a1. Replaced the positional indices with named constants, so reordering `allKnownSuggestions` can no longer attach the wrong suggestion to a coordinator condition — `build()` appends `emailSuggestion`/`calendarSuggestion`/`goalSuggestion`/etc. by name. `allKnownSuggestions` is now assembled from the same constants, and a new test asserts it equals exactly those named suggestions in order.

File: desktop/Desktop/Tests/ChatDiscoverabilityTests.swift [hunk 3]
```diff
@@ -105,6 +105,69 @@ final class ChatDiscoverabilityTests: XCTestCase {
...
+    }
+
+    @MainActor
+    func testOnboardingPromptsVocabularyIsBoundedAtSix() {
+        XCTAssertLessThanOrEqual(
+            OnboardingPromptSuggestionBuilder.allKnownSuggestions.count,
+            6,
+            "build() truncates with .prefix(6); any vocabulary entry beyond 6 becomes unreachable in some onboarding paths"
+        )
+    }
```
[the remark this answers is not in the document]
Comment (eulicesl) [reply desktop/Desktop/Tests/ChatDiscoverabilityTests.swift:183]: Fixed in fac5244a1. Changed the assertion to `== 6` (encoding both bounds), and — more fundamentally — `build()` no longer depends on the array length at all since it uses named constants instead of index access, so a shrunk vocabulary can't crash production regardless.

**Comment by eulicesl** [comment]
Closing this one too — it's effectively a test-only guardrail. The named-constant extraction is a behavior-preserving refactor (`build()` emits the same suggestions it did with the inline literals) and `allKnownSuggestions` exists only for the test, so there's no must-have functional change. Keeping test scaffolding out of upstream per the project's convention. Thanks!
